### PR TITLE
New chunkloading system

### PR DESCRIPTION
**Pull request by muradakh**
replace vehicles with a globally managed system

### CODE REVIEW

**Comment by EDToaster** [comment]
the wrench waypoint gives me an idea -- maybe we can render locomotive pathfinding decisions when the player has the wrench out

File: src/main/java/dev/murad/shipping/ShippingConfig.java [hunk 1]
```diff
@@ -77,25 +77,38 @@ public static class Server {
         public static final ForgeConfigSpec.ConfigValue<Integer> VESSEL_CHARGER_BASE_CAPACITY;
         public static final ForgeConfigSpec.ConfigValue<Integer> VESSEL_CHARGER_BASE_MAX_TRANSFER;
 
-        public static final ForgeConfigSpec.ConfigValue<List<String>> TRAIN_EXEMPT_DAMAGE_SOURCES;
-        public static final ForgeConfigSpec.ConfigValue<List<String>> VESSEL_EXEMPT_DAMAGE_SOURCES;
+        public static final ForgeConfigSpec.ConfigValue<List<? extends String>> TRAIN_EXEMPT_DAMAGE_SOURCES;
```
Comment (EDToaster) [inline src/main/java/dev/murad/shipping/ShippingConfig.java:80]: isn't java  string final? I dont understand the need for `? extends String`

File: src/main/java/dev/murad/shipping/entity/accessor/EnergyHeadVehicleDataAccessor.java [hunk 2]
```diff
@@ -17,69 +19,31 @@ public EnergyHeadVehicleDataAccessor(ContainerData data) {
...
-            this.arr.set(0, uuid);
+    public static class Builder extends HeadVehicleDataAccessor.Builder {
+        public Builder() {
+            this.arr = new SupplierIntArray(20);
         }
 
         public Builder withEnergy(IntSupplier energy) {
-            this.arr.setSupplier(1, () -> energy.getAsInt() & SHORT_MASK);
-            this.arr.setSupplier(2, () -> (energy.getAsInt() >> 16) & SHORT_MASK);
+            this.arr.setSupplier(15, () -> energy.getAsInt() & SHORT_MASK);
```
Comment (EDToaster) [inline src/main/java/dev/murad/shipping/entity/accessor/EnergyHeadVehicleDataAccessor.java:39]: kinda confused, why is this now higher?

File: src/main/java/dev/murad/shipping/util/EnrollmentHandler.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    public void save(CompoundTag tag){
+        if(uuid != null) {
+            tag.putUUID(UUID_TAG, uuid);
+        }
+    }
+
+    public void load(CompoundTag tag){
+        if(tag.contains(UUID_TAG)) {
+            uuid = tag.getUUID(UUID_TAG);
+            enrollMe = 5;
```
Comment (EDToaster) [inline src/main/java/dev/murad/shipping/util/EnrollmentHandler.java:61]: what's 5?

Comment (muradakh) [reply src/main/java/dev/murad/shipping/util/EnrollmentHandler.java:61]: Cooldown for enrollment attempts

File: src/main/java/dev/murad/shipping/entity/accessor/EnergyHeadVehicleDataAccessor.java (hunk 2 above)
Comment (muradakh) [reply src/main/java/dev/murad/shipping/entity/accessor/EnergyHeadVehicleDataAccessor.java:39]: Decided to have all the common attributes go first, and move energy/steam specific ones further out for future expansion. Got tired of refactoring everything every time we add something

**Review by EDToaster — APPROVED** [review]

**Pull request merged**